content/kubermatic/main/architecture/iam-role-based-access-control/_index.en.md

```diff
@@ -0,0 +1,121 @@
++++
+title = "IAM and Role-Based Access Control"
+date = 2020-04-02T12:07:15+02:00
+weight = 5
+
++++
+
+KKP components use OpenID Connect (OIDC) protocol for authentication and RBAC for authorization.
+By default, KKP provides [Dex](#authentication-with-dex) as OIDC provider, but you can configure your own provider. For more information,
+please refer to the [OIDC provider]({{< ref "../../tutorials-howtos/OIDC-Provider-Configuration" >}}) chapter.
+
+## Authentication with Dex
+[Dex](https://dexidp.io/) is an identity service that uses OIDC to drive authentication for KKP components. It acts as a
+portal to other identity providers through [connectors](https://dexidp.io/docs/connectors/). This lets Dex defer
+authentication to these connectors. Multiple connectors may be configured at the same time. Most popular are:
+* [GitHub](https://dexidp.io/docs/connectors/github/)
+* [Google](https://dexidp.io/docs/connectors/google/)
+* [LDAP](https://dexidp.io/docs/connectors/ldap/)
+* [Microsoft](https://dexidp.io/docs/connectors/microsoft/)
+* [OAuth 2.0](https://dexidp.io/docs/connectors/oauth/)
+* [OpenID Connect](https://dexidp.io/docs/connectors/oidc/)
+* [SAML2.0](https://dexidp.io/docs/connectors/saml/)
+
+Check out the [Dex documentation](https://dexidp.io/docs/connectors/) for a list of available providers and how to setup their configuration.
+
+To configure Dex connectors, edit `.dex.connectors` in the `values.yaml`
+
+Example to update or set up Github connector:
+```
+dex:
+    ingress:
+    [...]
+    connectors:
+    - type: github
+        id: github
+        name: GitHub
+        config:
+        clientID: <client_id>
+        clientSecret: <client_secret>
+        redirectURI: https://<your-kkp.domain>/dex/callback
+        orgs:
+        - name: <your-github-org>
+```
+
+And apply the changes to the cluster:
+
+```bash
+helm --namespace oauth upgrade --install --wait --values /path/to/your/helm-values.yaml oauth charts/oauth/
+```
```

Comment:
Honestly, we should probably not recommend this. Instead, people should use the `kubermatic-installer` to update their installation, no?

Reply:
done :)